### PR TITLE
IPFS repo migration 6 -> 7

### PR DESCRIPTION
**Pull request by cpacia**

### CODE REVIEW

File: repo/migrations/helpers/Migration020/record.proto [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+// 1. Protobuf binary (tested with protoc 3.0.0). - https://github.com/gogo/protobuf/releases
+// 2. Gogo Protobuf (tested with gogo 0.3). - https://github.com/gogo/protobuf
+// Now from `libp2p/go-libp2p-record/pb` you can run...
+// `protoc --gogo_out=. record.proto`
+
+syntax = "proto2";
+
+// RecordOldFormat represents a dht record that contains a value
+// for a key value pair
+message RecordOldFormat {
```
Comment (placer14) [inline repo/migrations/helpers/Migration020/record.proto]: Appreciate your consideration here. 🤝

File: repo/migrations/Migration020.go [hunk 2]
```diff
@@ -0,0 +1,342 @@
...
+		return fmt.Errorf("bumping repover to 21: %s", err.Error())
+	}
+	return nil
+}
+
+func (Migration020) Down(repoPath string, dbPassword string, testnet bool) error {
+	// We're downgrading from version 7.
+	fsrepo.RepoVersion = 7
+	r, err := fsrepo.Open(repoPath)
+	fsrepo.RepoVersion = 6
```
Comment (placer14) [inline repo/migrations/Migration020.go]: What does setting these values do?

File: repo/migrations/Migration020.go [hunk 3]
```diff
@@ -0,0 +1,342 @@
...
+
+// RecordBlobForSig returns the blob protected by the record signature
+func RecordBlobForSig(r *dhtpb.RecordOldFormat) []byte {
+	k := []byte(r.GetKey())
+	v := r.GetValue()
+	a := []byte(r.GetAuthor())
+	return bytes.Join([][]byte{k, v, a}, []byte{})
+}
+
+func IpnsKeysForID(id peer.ID) (name, ipns string) {
```
Comment (placer14) [inline repo/migrations/Migration020.go]: Minor: Capitalize Acronyms: `IPNSKeysForID`

File: repo/migrations/Migration020.go [hunk 4]
```diff
@@ -0,0 +1,342 @@
...
+	return nil
+}
+
+func myKey(r repo.Repo) (ci.PrivKey, error) {
+	cfg, err := r.Config()
+	if err != nil {
+		return nil, err
+	}
+
+	sk, err := cfg.Identity.DecodePrivateKey("passphrase todo!")
```
Comment (placer14) [inline repo/migrations/Migration020.go]: Forgotten todo here?

File: repo/migrations/helpers/Migration020/record.proto (hunk 1 above)
Comment (cpacia) [reply repo/migrations/helpers/Migration020/record.proto]: Sure I can do that

File: repo/migrations/Migration020.go (hunk 2 above)
Comment (cpacia) [reply repo/migrations/Migration020.go]: They are copied from the ipfs migration. I assume they are needed.

File: repo/migrations/Migration020.go (hunk 4 above)
Comment (cpacia) [reply repo/migrations/Migration020.go]: This was also in their migration code ¯\_(ツ)_/¯ afaict the password isn't used.

**Pull request merged**